lib/iris/tests/unit/cube/test_CubeList.py

```diff
@@ -341,15 +341,21 @@ def test_pass(self):
         self.cubelist[:2] = (self.cube2, self.cube3)
         self.assertEqual(
             self.cubelist,
-            iris.cube.Cubelist([self.cube2, self.cube3, self.cube1]))
+            iris.cube.CubeList([self.cube2, self.cube3, self.cube1]))
 
     def test_fail(self):
-        msg = ("Elements of cubelists must be Cube instances.  Got "
-               "<class 'NoneType'>.")
+        msg = "Elements of cubelists must be Cube instances.  Got "
         with self.assertRaisesRegexp(TypeError, msg):
             self.cubelist[0] = None
-        with self.assertRaisesRegexp(TypeError, msg):
+        msg = "Elements of cubelists must be Cube instances."
+        with self.assertRaisesRegexp(ValueError, msg):
             self.cubelist[0:2] = [self.cube3, None]
```

Comment:
This `ValueError` is not raised under Python2.  If I've understood [this StackOverflow question](https://stackoverflow.com/questions/32772755/how-do-getitem-setitem-work-with-slices) correctly, we'd need to implement a `__setslice__` method for Python2. :open_mouth: